### PR TITLE
Porting okhttp3 integration to Glide 3

### PR DESCRIPTION
**Pull request by frindly**
I ported the okhttp3 integration to the Glide 3 branch.

### CODE REVIEW

**Comment by frindly** [comment]
I signed the CLA.

**Comment by TWiStErRob** [comment]
Lgtm.

Reminder: #257 will need a re-fix for this as well.

**Comment by TWiStErRob** [comment]
Oh and I opened #888 because the version looks weird compared to the v4 okhttp3 one. @frindly you needn't worry about this.

**Comment by frindly** [comment]
Thanks -- I couldn't really tell what the version strategy was, so I tried to be consistent with what was in the other integrations.

**Comment by TWiStErRob** [comment]
Yes, you did good. The current idea is that all integration libs have the same version, separate from the main lib's version (this might change with 4.0). I tried to make it more obvious by removing duplication, but it didn't work, resulting in [GRADLE-3380](https://issues.gradle.org/browse/GRADLE-3380).

**Comment by sjudd** [comment]
LGTM, thanks! Would you mind squashing your commits? If not I'll get to it soon.

**Comment by frindly** [comment]
squashed

**Comment by sjudd** [comment]
Awesome thanks!

**Comment by TWiStErRob** [comment]
@frindly you may be interested in #897

**Comment by frindly** [comment]
Of course -- they upgrade off RC one day later.  I'll look at updating it next week.

**Comment by b95505017** [comment]
Glide 3 using sync call with OkHttp without callback, so I think it could upgrade the version directly.

**Comment by b95505017** [comment]
Would you mind publish a new stable release for Glide 3 with OkHttp 3.0 integration?

File: integration/okhttp3/build.gradle [hunk 1]
```diff
@@ -0,0 +1,26 @@
+apply plugin: 'com.android.library'
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    compile project(':library')
```
Comment (TWiStErRob) [inline integration/okhttp3/build.gradle]: I think we missed this, see #941
Library module in v3 is not published, everything there depends on `:glide` as far as I remember.

**Pull request merged**